salt/utils/master.py

```diff
@@ -214,16 +214,16 @@ def _get_minion_pillar(self, *minion_ids, **kwargs):
             cret = dict([(minion_id, mcache) for (minion_id, mcache) in six.iteritems(cached_pillar) if mcache])
             missed_minions = [minion_id for minion_id in minion_ids if minion_id not in cret]
             log.debug('Missed cached minion pillars for: %s', missed_minions)
-            if self.pillar_fallback:
+            if self.pillar_fallback and missed_minions:
                 lret = dict([(minion_id, self._get_live_minion_pillar(minion_id, grains.get(minion_id, {}))) for minion_id in missed_minions])
-            ret = dict(list(six.iteritems(dict([(minion_id, {}) for minion_id in minion_ids]))) + list(lret.items()) + list(cret.items()))
+            ret = {key: value for key, value in [(minion_id, {}) for minion_id in minion_ids] + list(cret.items()) + list(lret.items())}
         else:
             lret = dict([(minion_id, self._get_live_minion_pillar(minion_id, grains.get(minion_id, {}))) for minion_id in minion_ids])
             missed_minions = [minion_id for minion_id in minion_ids if minion_id not in lret]
             log.debug('Missed live minion pillars for: %s', missed_minions)
-            if self.pillar_fallback:
+            if self.pillar_fallback and missed_minions:
                 cret = dict([(minion_id, mcache) for (minion_id, mcache) in six.iteritems(cached_pillar) if mcache])
-            ret = dict(list(six.iteritems(dict([(minion_id, {}) for minion_id in minion_ids]))) + list(lret.items()) + list(cret.items()))
+            ret = {key: value for key, value in [(minion_id, {}) for minion_id in minion_ids] + list(cret.items()) + list(lret.items())}
```

Comment:
Same as the previous comment, can you please switch to using `list(six.iteritems(dictname))` here and on line 219?